CMakeLists.txt

```diff
@@ -278,7 +278,16 @@ if(PYBIND11_INSTALL)
 
   # pkg-config support
   if(NOT prefix_for_pc_file)
-    set(prefix_for_pc_file "${CMAKE_INSTALL_PREFIX}")
+    if(IS_ABSOLUTE "${CMAKE_INSTALL_DATAROOTDIR}")
+      set(prefix_for_pc_file "${CMAKE_INSTALL_PREFIX}")
+    else()
+      set(pc_datarootdir "${CMAKE_INSTALL_DATAROOTDIR}")
+      set(prefix_for_pc_file "\${pcfiledir}/..")
+      while(pc_datarootdir)
```

Comment:
cmake_path in CMake 3.20 would make this so much nicer. But sadly we won't depend on that for a while.

Reply:
Yep. I'm still targeting 3.12 in my main projects. Can't wait for folks to get onto the next LTS :) .